tiledb/sm/query/readers/sparse_index_reader_base.cc

```diff
@@ -553,16 +575,51 @@ Status SparseIndexReaderBase::apply_query_condition(
             rt->count_cells();
           }
 
-          // Compute the result of the query condition for this tile.
-          if (!condition_.empty()) {
+          // Make sure we have a condition bitmap if needed.
+          if (!condition_.empty() || !delete_conditions_.empty()) {
             rt->ensure_bitmap_for_query_condition();
+          }
+
+          // Compute the result of the query condition for this tile
+          if (!condition_.empty()) {
             RETURN_NOT_OK(condition_.apply_sparse<BitmapType>(
                 *(frag_meta->array_schema().get()), *rt, rt->bitmap_with_qc()));
             if (array_schema_.allows_dups()) {
               rt->count_cells();
             }
           }
 
+          // Apply delete conditions.
+          if (!delete_conditions_.empty()) {
+            for (uint64_t i = 0; i < delete_conditions_.size(); i++) {
+              auto delete_timestamp =
+                  delete_conditions_[i].condition_timestamp();
+
+              // Check the delete condition timestamp is after the fragment
+              // start.
+              if (delete_timestamp >= frag_meta->timestamp_range().first) {
+                // Apply timestamped condition or regular condition.
+                if (!frag_meta->has_timestamps() ||
+                    delete_timestamp > frag_meta->timestamp_range().second) {
```

Comment:
File shortcut item to disallow deletes in between old style consolidated fragments.